proc/put/put.go

```diff
@@ -81,116 +84,284 @@ func (p *Proc) eval(in *zng.Record) ([]zng.Value, error) {
 	return vals, nil
 }
 
-func (p *Proc) buildRule(inType *zng.TypeRecord, vals []zng.Value) (*putRule, error) {
-	n := len(inType.Columns)
-	cols := make([]zng.Column, n, n+len(p.clauses))
-	copy(cols, inType.Columns)
-	clauseTypes := make([]clauseType, len(p.clauses))
-	var nreplace int
-	replace := make([]int, n)
-	for k := range replace {
-		replace[k] = -1
+// A op is a recursive data structure encoding a series of steps to be
+// carried out to construct an output record from an input record and
+// a slice of evaluated clauses.
+type op struct {
+	op        step
```

Comment:
Swapping these type names would make this so much clearer.
```suggestion
type step struct {
	op        op
```

Reply:
Agreed, done in 00ad1f26035b85c814ebd512daaad2b39e14c7aa
(I went back and forth and had that at some point (as reflected by `func (s op)`) but got stuck in a local minima. )